src/js/settings.js

```diff
@@ -591,33 +591,145 @@ function exportSubscriptions() {
   const dateYear = date.getFullYear();
   const dateString = 'freetube-subscriptions-' + dateYear + '-' + dateMonth + '-' + dateDay;
 
-  // Open user file browser. User gives location of file to be created.
+    switch(document.querySelector('#exportSelect').value){
+      
+      case "NewPipe":
+        exportNewpipeSubscriptions(dateYear, dateMonth, dateDay); 
+      break;
+      case "OPML":
+        exportOpmlSubscriptions(dateYear, dateMonth, dateDay);
+      break;
+      default:
+        // Open user file browser. User gives location of file to be created.
+        dialog.showSaveDialog({
+          defaultPath: dateString,
+          filters: [{
+            name: 'Database File',
+            extensions: ['db']
+          }, ]
+        }, function(fileLocation) {
+          console.log(fileLocation);
+          if (typeof(fileLocation) === 'undefined') {
+            console.log('Export Aborted');
+            return;
+          }
+          fs.readFile(appDatabaseFile, function(readErr, data) {
+            if (readErr) {
+              throw readErr;
+            }
+            fs.writeFile(fileLocation, data, function(writeErr) {
+              if (writeErr) {
+                throw writeErr;
+              }
+              showToast('Susbcriptions have been successfully exported');
+            });
+          })
+        });
+    }
+}
+/**
+ * Export the subscriptions database compatable with NewPipe.
+ *
+ * @return {Void}
+ */
+function exportNewpipeSubscriptions(dateYear, dateMonth, dateDay){
+
+  const dateString = 'newpipe-subscriptions-' + dateYear + '-' + dateMonth + '-' + dateDay;
+
   dialog.showSaveDialog({
     defaultPath: dateString,
     filters: [{
-      name: 'Database File',
-      extensions: ['db']
+      name: 'JSON',
+      extensions: ['json']
     }, ]
   }, function(fileLocation) {
     console.log(fileLocation);
     if (typeof(fileLocation) === 'undefined') {
       console.log('Export Aborted');
       return;
     }
-    fs.readFile(appDatabaseFile, function(readErr, data) {
-      if (readErr) {
-        throw readErr;
-      }
-      fs.writeFile(fileLocation, data, function(writeErr) {
-        if (writeErr) {
-          throw writeErr;
+    returnSubscriptions().then((result)=>{
+        let newpipe = {
+          app_version: "0.15.1",
```

Comment:
NewPipe's current version is 16.1.  Let's match that version number.

Reply:
can be done. I intentionally used 0.15.1 for backward compatibility

Reply:
Is there any notable changes that would require the version number to be lower for backwards compatibility reasons?  If there isn't, let's match their version number.

Reply:
Nothing ATM